tests/cases/unittests/tsserverProjectSystem.ts

```diff
@@ -188,6 +212,26 @@ namespace ts {
             };
         }
 
+        triggerDirectoryWatcherCallback(directoryName: string, fileName: string): void {
+            const path = this.toPath(directoryName);
+            const callbacks = lookUp(this.watchedDirectories, path);
+            if (callbacks) {
+                for (const callback of callbacks) {
+                    callback.cb(fileName);
+                }
+            }
+        }
+
+        triggerFileWatcherCallback(fileName: string): void {
+            const path = this.toPath(fileName);
+            const callbacks = lookUp(this.watchedFiles, path);
+            if (callbacks) {
+                for (const callback of callbacks) {
+                    callback(path, /*removed*/ true);

```

Comment:
do we ever use `removed` parameter? As I can see we already double check the presence of file in event handler so we should either remove it if we think it should not be trusted or switch to using it (and in this case it should become parameter in `triggerFileWatcherCallback`)?